extensions/src/AWSSDK.Extensions.NETCore.Setup/ConfigurationExtensions.cs

```diff
@@ -55,33 +67,10 @@ public static AWSOptions GetAWSOptions(this IConfiguration config)
         /// <param name="configSection">The config section to extract AWS options from.</param>
         /// <returns>The AWSOptions containing the values set in configuration system.</returns>
         public static AWSOptions GetAWSOptions(this IConfiguration config, string configSection)
-        {
-            return GetAWSOptions<DefaultClientConfig>(config, configSection);
-        }
-
-        /// <summary>
-        /// Constructs an AWSOptions class with the options specified in the "AWS" section in the IConfiguration object.
-        /// </summary>
-        /// <typeparam name="TConfig">The AWS client config to be used in creating clients, like AmazonS3Config.</typeparam>
-        /// <param name="config"></param>
-        /// <returns>The AWSOptions containing the values set in configuration system.</returns>
-        public static AWSOptions GetAWSOptions<TConfig>(this IConfiguration config) where TConfig : ClientConfig, new()
-        {
-            return GetAWSOptions<TConfig>(config, DEFAULT_CONFIG_SECTION);
-        }
-
-        /// <summary>
-        /// Constructs an AWSOptions class with the options specified in the "AWS" section in the IConfiguration object.
-        /// </summary>
-        /// <typeparam name="TConfig">The AWS client config to be used in creating clients, like AmazonS3Config.</typeparam>
-        /// <param name="config"></param>
-        /// <param name="configSection">The config section to extract AWS options from.</param>
-        /// <returns>The AWSOptions containing the values set in configuration system.</returns>
-        public static AWSOptions GetAWSOptions<TConfig>(this IConfiguration config, string configSection) where TConfig : ClientConfig, new()
```

Comment:
The generic overloads passing in the type of service client config got removed because `DefaultClientConfig` no longer extends from ClientConfig. The purpose of these overloads were for later parts in the library when constructing the service client to have the service client config specific be passed in here so that service specific settings could be set. This now accomplished differently using the `ServiceSpecificSettings` property on `DefaultClientConfig` and the `ProcessServiceSpecificSettings` method above for using reflection to copy the values over.